index.d.ts

```diff
@@ -0,0 +1,373 @@
+// TypeScript Version: 4.2
+
+export type ImporterReturnType =
+  | {file: string}
+  | {contents: string}
+  | Error
+  | null;
+
+export type Importer = (
+  url: string,
+  prev: string,
+  done: (data: ImporterReturnType) => void
+) => ImporterReturnType | void;
+
+export interface LogData {
+  /**
+   * The message as-is in the file/data
+   */
+  message: string;
+
+  /**
+   * A set of information related to the location of the error
+   */
+  source: SourceSpan;
+
+  /**
+   * Is it a deprecation message
+   */
+  deprecation: boolean;
+}
+
+export interface Logger {
+  /**
+   * A function that is called each time `@warn` is encountered
+   * @param data All data relative to the `@warn` line
+   */
+  warn?(data: LogData): void;
+
+  /**
+   * A function that is called each time `@debug` is encountered
+   * @param data All data relative to the `@debug` line
+   */
+  debug?(data: LogData): void;
+}
+
+export interface Options {
+  /**
+   * Path to a file to compile.
+   *
+   * @default null
+   */
+  file?: string;
+
+  /**
+   * A string to pass to compile.
+   *
+   * It is recommended that you use `includePaths` in conjunction with this so that sass can find files when using the @import directive.
+   *
+   * @default null
+   */
+  data?: string;
```

Comment:
Although each of these is individually optional, at least one of them and optionally both *must* be passed by the caller. You can see [here](https://github.com/sass/embedded-host-node/blob/661fa06465e8cae1ecc518e11dc6dae24dccb573/lib/src/node-sass/render.ts#L13-L44) how @awjin represents this for the Node embedded host.

Reply:
Thanks for the sample - every day is a learning day 😁